sklearn/cross_validation.py

```diff
@@ -1185,8 +1185,10 @@ def _fit_and_predict(estimator, X, y, train, test, verbose, fit_params):
     # Adjust length of sample weights
     n_samples = _num_samples(X)
     fit_params = fit_params if fit_params is not None else {}
-    fit_params = dict([(k, np.asarray(v)[train]
-                       if hasattr(v, '__len__') and len(v) == n_samples else v)
+    fit_params = dict([(k, (safe_indexing(v.tocsr(), train) if sp.issparse(v)
+                            else np.asarray(v)[train])
+                        if _is_arraylike(v) and _num_samples(v) == n_samples
+                        else v)
```

Comment:
It seems to be properly tested [here] (https://github.com/scikit-learn/scikit-learn/blob/79eabdda25b5f4bef6b03da129489318cac54ac0/sklearn/tests/test_cross_validation.py#L603)
No i don't think that should be `(k, v)` It is similar to the code below
```
>>> a = {'a':42}
>>> b = dict([(k, ((v+2 if v<10 else v+3) if v>20 else v)) for k,v in a.items()])
>>> b
{'a': 45}
```